DashAI/front/src/components/datasets/DatasetSummaryStep.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useState } from "react";
+import { useSnackbar } from "notistack";
+import { Paper, Grid, Typography } from "@mui/material";
+import PropTypes from "prop-types";
+import { DataGrid, useGridApiContext } from "@mui/x-data-grid";
+import uuid from "react-uuid";
+import {
+  getDatasetSample as getDatasetSampleRequest,
+  getDatasetTypes as getDatasetTypesRequest,
+} from "../../api/datasets";
+import { dataTypesList, columnTypesList } from "../../utils/typesLists";
+import SelectTypeCell from "./SelectTypeCell";
+function DatasetSummaryStep({
+  uploadedDataset,
+  setNextEnabled,
+  datasetUploaded,
+  updateColumnTypes,
+  setUpdateColumnTypes,
+}) {
+  const [loading, setLoading] = useState(true);
+  const { enqueueSnackbar } = useSnackbar();
+  const [rows, setRows] = useState([]);
+
+  const getDatasetInfo = async () => {
+    setLoading(true);
+    try {
+      const dataset = await getDatasetSampleRequest(uploadedDataset.id);
+      const types = await getDatasetTypesRequest(uploadedDataset.id);
+      const rowsArray = Object.keys(dataset).map((name) => {
+        return {
+          id: uuid(),
+          columnName: name,
+          example: dataset[name][0],
+          columnType: types[name].type,
+          dataType: types[name].dtype,
+        };
+      });
```

Comment:
Why not to use an enumeration?

```suggestion
      const rowsArray = Object.keys(dataset).map((name, idx) => {
        return {
          id: idx,
          columnName: name,
          example: dataset[name][0],
          columnType: types[name].type,
          dataType: types[name].dtype,
        };
      });
```